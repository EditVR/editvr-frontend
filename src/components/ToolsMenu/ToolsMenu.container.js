/**
 * @file ToolsMenu.container.js
 * Exports a redux-connected ToolsMenu component.
 */

import { withRouter } from 'react-router-dom';

import ToolsMenu from './ToolsMenu';

export default withRouter(ToolsMenu);
