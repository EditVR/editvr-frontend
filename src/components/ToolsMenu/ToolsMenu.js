/**
 * @file ToolsMenu.js
 * Exports a component that renders EditVR's tools.
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { withStyles, Button, Tooltip } from '@material-ui/core';
import { OpenWith, TouchApp, PanTool } from '@material-ui/icons';

import {
  MODE_COMPONENT_SELECTING,
  MODE_COMPONENT_PLACING
} from '../../constants';
import ToolsMenuStyles from './ToolsMenu.style';

const ToolsMenu = ({
  classes,
  history: {
    location: { pathname: location }
  },
  match: {
    params: { experienceSlug, sceneSlug }
  }
}) => {
  const basePath = `/experience/vreditor/${experienceSlug}`;
  const previewPath = `${basePath}/${sceneSlug !== 'scene' ? sceneSlug : ''}`;
  const selectingPath = `${basePath}/${sceneSlug}/${MODE_COMPONENT_SELECTING}`;
  const placingPath = `${basePath}/${sceneSlug}/${MODE_COMPONENT_PLACING}`;

  if (!sceneSlug || sceneSlug === 'scene') {
    return null;
  }

  return (
    <Fragment>
      <Tooltip title="Preview" placement="right-start">
        <Button
          className={classes.toolButton}
          variant="fab"
          component={Link}
          disabled={location === previewPath}
          color="primary"
          to={previewPath}
        >
          <OpenWith />
        </Button>
      </Tooltip>
      <Tooltip title="Select components for editing" placement="right-start">
        <Button
          className={classes.toolButton}
          variant="fab"
          component={Link}
          disabled={location === selectingPath}
          color="primary"
          to={selectingPath}
        >
          <TouchApp />
        </Button>
      </Tooltip>
      <Tooltip
        title="Position components by dragging and dropping them"
        placement="right-start"
      >
        <Button
          className={classes.toolButton}
          variant="fab"
          component={Link}
          disabled={location === placingPath}
          color="primary"
          to={placingPath}
        >
          <PanTool />
        </Button>
      </Tooltip>
    </Fragment>
  );
};

ToolsMenu.propTypes = {
  classes: PropTypes.shape({
    toolButton: PropTypes.string.isRequired
  }).isRequired,
  history: PropTypes.shape({
    location: PropTypes.shape({
      pathname: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      experienceSlug: PropTypes.string.isRequired,
      sceneSlug: PropTypes.string
    }).isRequired
  }).isRequired
};

ToolsMenu.defaultProps = {};

export default withStyles(ToolsMenuStyles)(ToolsMenu);
