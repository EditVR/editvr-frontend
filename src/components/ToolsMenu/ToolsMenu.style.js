/**
 * @file ToolsMenu.style.js
 * Exports ToolsMenu component styles.
 */

export default theme => ({
  toolButton: {
    display: 'block',
    marginTop: theme.spacing.unit * 2,
    textAlign: 'center',
    paddingTop: theme.spacing.unit * 2
  }
});
