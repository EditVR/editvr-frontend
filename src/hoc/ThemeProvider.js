/**
 * @file ThemeProvider.js
 * Exports a HOC ThemeProvider component that attaches children their MUI theme.
 */

import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';

import muiTheme from '../styles/muiTheme';

const ThemeProvider = WrappedComponent => props => (
  <MuiThemeProvider theme={muiTheme}>
    <WrappedComponent {...props} />
  </MuiThemeProvider>
);

export default ThemeProvider;
