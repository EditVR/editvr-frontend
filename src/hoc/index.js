/**
 * @file index.js
 * Exports all hoc components.
 */

import ThemeProvider from './ThemeProvider';

export { ThemeProvider };
