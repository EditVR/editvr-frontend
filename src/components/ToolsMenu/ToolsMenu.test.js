/**
 * @file ToolsMenu.test.js
 * Contains tests for ToolsMenu.js.
 */

import React from 'react';
import { MemoryRouter as Router, Route } from 'react-router-dom';
import { createMount } from '@material-ui/core/test-utils';

import { ThemeProvider } from '../../hoc';
import ToolsMenu from './ToolsMenu.container';

const ThemedToolsMenu = ThemeProvider(ToolsMenu);

describe('<ToolsMenu />', () => {
  it('Matches its snapshot', () => {
    expect(
      createMount()(
        <Router initialEntries={['/experience/vreditor/test/test']}>
          <Route
            path="/experience/vreditor/:experienceSlug/:sceneSlug?"
            component={ThemedToolsMenu}
          />
        </Router>
      ).html()
    ).toMatchSnapshot();
  });
});
