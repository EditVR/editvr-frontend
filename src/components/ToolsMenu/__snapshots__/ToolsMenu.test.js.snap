// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolsMenu /> Matches its snapshot 1`] = `"<a tabindex=\\"-1\\" class=\\"MuiButtonBase-root-26 MuiButtonBase-disabled-27 MuiButton-root-10 MuiButton-raised-16 MuiButton-fab-21 MuiButton-raisedPrimary-17 MuiButton-disabled-20 ToolsMenu-toolButton-1\\" role=\\"button\\" href=\\"/experience/vreditor/test/test\\"><span class=\\"MuiButton-label-11\\"><svg class=\\"MuiSvgIcon-root-29\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><g><path d=\\"M10 9h4V6h3l-5-5-5 5h3v3zm-1 1H6V7l-5 5 5 5v-3h3v-4zm14 2l-5-5v3h-3v4h3v3l5-5zm-9 3h-4v3H7l5 5 5-5h-3v-3z\\"></path></g></svg></span></a>"`;
